recipes/orsopy/meta.yaml

```diff
@@ -0,0 +1,44 @@
+{% set name = "orsopy" %}
+{% set version = "1.1.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/orsopy-{{ version }}.tar.gz
+  sha256: fb3c77197987dac61abf16a404bfc02474cf35a0994dd5dc73b956950f0c85de
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >=3.8
+    - pip
+  run:
+    - python >=3.8
+    - pyyaml >=5.4.1
+
+test:
+  imports:
+    - orsopy
+  commands:
+    - pip check
```

Comment:
@bmaranville, the testing should be quite easy. All we should have to do is add something like `pytest --pyargs orsopy` here. However, we'd have to make another orsopy release to take into account that the tests are now within the package structure.

The `imports` section above is a really basic smoke test to see if you can import orsopy.

This recipe gets auto updated when a bot detects that another release is available on PyPI. This recipe got generated by a conda-forge package called `grayskull`, so I didn't need to actually write anything.

You can checkout the refnx-feedstock [here](https://github.com/conda-forge/refnx-feedstock/blob/main/recipe/meta.yaml).

Reply:
What I don't understand is how to make sure that the conda-forge tests will pass *before* you make a release on the original repo... otherwise it seems like the conda-forge release doesn't work

Reply:
Making a conda-forge release is more complicated for packages that have native code, because the actual build/compilation of that native code could fail. That's not a factor for pure Python packages, such as orsopy.

For packages that have native code it's good to have detailed test suite on the original repo. If you're testing across linux/macOS/win then you're already checking the build and any test dependence on platforms are picked up early. This means it's unlikely that a conda-forge build will fail.

If there are failed tests (but the compilation completes) in a conda-forge build process then they're likely to be present in the PyPI wheel as well, it's just they weren't picked up by CI in the original repo. If that's the case one can either comment out the failing test in the conda-forge recipe or add a patch (e.g. [here](https://github.com/conda-forge/staged-recipes/pull/19166/commits/0284fc6da273031a4f93a1fea4533822cd4b385d))

A further alternative is to build the conda-forge recipe as part of the orsopy CI.